src/modules/storefront/components/Products.tsx

```diff
@@ -0,0 +1,202 @@
+import { CSSProperties, useEffect } from 'react';
+
+import { Autoplay, Navigation } from 'swiper';
+import { Swiper, SwiperSlide } from 'swiper/react';
+
+import { Product, ProductsData, ProductsDefault } from '../interfaces';
+
+import 'swiper/css';
+import 'swiper/css/navigation';
+
+import './Products.css';
+
+const getCardInfo = (
+  data: ProductsData['card'],
+  defaultData: ProductsDefault['card']
+): ProductsDefault['card'] => {
+  return {
+    name: data?.name !== undefined ? data.name : defaultData.name,
```

Comment:
Essa questão do default Theme e do thema depois para mim não faz sentido.... pq sempre tem a validação e ambos na teoria podem vir undefined se o cara não setar nada.... isso para todos os componentes 
oq eu faria seria deixar a interface de ambos iguais setando fazendo um spread... {...defaultTheme, ...Theme} e ai tudo que fosse undefined de ambos eu faria uma tratativa para um valor default que podemos definir com o marquinhos. 
a situação que eu to falando é tipo, criei uma loja agora e o back não manda nada de informação para a gente, até pq não foi cadastrado, nesse caso o valor seria undefined de todas as propriedades css, e não sei qual seria o comportamento disso quando renderizasse...
Isso é só uma sugestão